backend/btrixcloud/emailsender.py

```diff
@@ -46,70 +47,92 @@ def __init__(self):
 
         self.log_sent_emails = is_bool(os.environ.get("LOG_SENT_EMAILS"))
 
-        self.templates = Jinja2Templates(
-            directory=os.path.join(os.path.dirname(__file__), "email-templates")
-        )
+        email_template_endpoint = os.environ.get("EMAIL_TEMPLATE_ENDPOINT")
+        if not email_template_endpoint:
+            raise ValueError(
+                "Environment variable EMAIL_TEMPLATE_ENDPOINT is required but not set"
+            )
+        self.email_template_endpoint = email_template_endpoint
 
-    def _send_encrypted(self, receiver: str, name: str, **kwargs) -> None:
+    async def _send_encrypted(self, receiver: str, name: str, **kwargs) -> None:
         """Send Encrypted SMTP Message using given template name"""
 
-        full = self.templates.env.get_template(name).render(kwargs)
-        parts = full.split("~~~")
-        if len(parts) == 3:
-            subject, html, text = parts
-        elif len(parts) == 2:
-            subject, text = parts
-            html = None
-        else:
-            raise HTTPException(status_code=500, detail="invalid_email_template")
-
-        if self.log_sent_emails:
-            print(full, flush=True)
-
-        if not self.smtp_server:
-            print(
-                f'Email: created "{name}" msg for "{receiver}", but not sent (no SMTP server set)',
-                flush=True,
-            )
-            return
-
-        msg: Union[EmailMessage, MIMEMultipart]
-
-        if html:
-            msg = MIMEMultipart("alternative")
-            msg.attach(MIMEText(text.strip(), "plain"))
-            msg.attach(MIMEText(html.strip(), "html"))
-        else:
-            msg = EmailMessage()
-            msg.set_content(text.strip())
-
-        msg["Subject"] = subject.strip()
-        msg["From"] = self.reply_to
-        msg["To"] = receiver
-        msg["Reply-To"] = msg["From"]
-
-        context = ssl.create_default_context()
-        with smtplib.SMTP(self.smtp_server, self.smtp_port) as server:
-            if self.smtp_use_tls:
-                server.ehlo()
-                server.starttls(context=context)
-            server.ehlo()
-            if self.password:
-                server.login(self.sender, self.password)
-            server.send_message(msg)
-            # server.sendmail(self.sender, receiver, message)
-
-    def send_user_validation(
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.post(
+                    self.email_template_endpoint + "/" + name,
+                    json=kwargs,
+                ) as resp:
+                    if resp.status != 200:
+                        raise HTTPException(
+                            status_code=resp.status,
+                            detail=await resp.text(),
+                        )
+
+                    json = await resp.json()
```

Comment:
Eventually, it might make sense to add a pydantic model for the response to aid with customization (if someone wanted to add their own template generator i guess), but not sure its worth doing now.